gnovm/pkg/gnolang/values.go

```diff
@@ -1010,6 +1010,28 @@ func (tv TypedValue) Copy(alloc *Allocator) (cp TypedValue) {
 	return
 }
 
+// DeepCopy makes of copy of the underlying value in the case of reference values.
```

Comment:
This is a misnomer. It doesn't do a deep copy, a deep copy should deep copy all children recursively. It only does a slightly deeper copy if tv happens to be a RefValue. But this function is useful at least in the context of uverse, so I would call this "unrefCopy" and not even expose it.

Reply:
unrefCopy makes *a* copy of the underlying value, but first unreferences if RefValue.